controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -65,47 +97,460 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+	if availableCephFileSystems == 0 {
+		return fmt.Errorf("No CephFileSystem found in the cluster")
+	}
+	if availableCephFileSystems > 1 {
+		klog.Warningf("More than one CephFileSystem found in the cluster, selecting the first one")
+	}
+
+	cephFileSystemName = cephFilesystemList.Items[0].Name
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephFilesystemSubVolumeGroup, func() error {
+		if err := r.own(r.cephFilesystemSubVolumeGroup); err != nil {
+			return err
+		}
+
+		r.cephFilesystemSubVolumeGroup.Spec = rookCephv1.CephFilesystemSubVolumeGroupSpec{
+			FilesystemName: cephFileSystemName,
+		}
+		return nil
+	})
+
+	if err != nil {
+		r.log.Error(
+			err,
+			"Failed to update CephFilesystemSubVolumeGroup.",
+			"CephFilesystemSubVolumeGroup",
+			klog.KRef(r.cephFilesystemSubVolumeGroup.Namespace, r.cephFilesystemSubVolumeGroup.Name),
+		)
+		return err
+	}
+
+	cephClients := map[string]string{
+		"provisioner": r.cephClientProvisioner.Name,
+		"node":        r.cephClientNode.Name,
+	}
+	phase := ""
+	if r.cephFilesystemSubVolumeGroup.Status != nil {
+		phase = string(r.cephFilesystemSubVolumeGroup.Status.Phase)
+	}
+
+	r.setCephResourceStatus(r.cephFilesystemSubVolumeGroup.Name, "CephFilesystemSubVolumeGroup", phase, cephClients)
+
+	return nil
+}
+
+func (r *StorageClassClaimReconciler) reconcileCephClientRBDProvisioner() error {
+
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"mgr": "allow rw",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientProvisioner, func() error {
+		if err := r.own(r.cephClientProvisioner); err != nil {
+			return err
+		}
+
+		addStorageRelatedAnnotations(r.cephClientProvisioner, r.storageClassClaim.Name, "rbd", "provisioner")
+		r.cephClientProvisioner.Spec = desired.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.log.Error(err,
+			"Failed to update CephClient.",
+			"CephClient",
+			klog.KRef(r.cephClientProvisioner.Namespace, r.cephClientProvisioner.Name),
+		)
+		return err
+	}
+
+	phase := ""
+	if r.cephClientProvisioner.Status != nil {
+		phase = string(r.cephClientProvisioner.Status.Phase)
+	}
+
+	r.setCephResourceStatus(r.cephClientProvisioner.Name, "CephClient", phase, nil)
+
+	return nil
+}
+
+func (r *StorageClassClaimReconciler) reconcileCephClientRBDNode() error {
+
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"mgr": "allow rw",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientNode, func() error {
+		if err := r.own(r.cephClientNode); err != nil {
+			return err
+		}
+
+		addStorageRelatedAnnotations(r.cephClientNode, r.storageClassClaim.Name, "rbd", "node")
+		r.cephClientNode.Spec = desired.Spec
```

Comment:
Same as last comment. Embed here (and remove from lines 394-402)
```suggestion
		r.cephClientNode.Spec = rookCephv1.ClientSpec{
			Caps: map[string]string{
				"mon": "profile rbd",
				"mgr": "allow rw",
				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
			},
		}
```

Reply:
ack